subsys/bluetooth/host/audio/iso.c

```diff
@@ -0,0 +1,1075 @@
+	__ASSERT(conn->type == BT_CONN_TYPE_ISO, "Invalid connection type");
+
+	if (!evt->status) {
+		/* TODO: Add CIG sync delay */
+		bt_conn_set_state(conn, BT_CONN_CONNECTED);
+		bt_conn_unref(conn);
+		return;
+	}
+
+	conn->err = evt->status;
+	bt_iso_disconnected(conn);
+	bt_conn_unref(conn);
+}
+
+int hci_le_reject_cis(uint16_t handle, uint8_t reason)
+{
+	struct bt_hci_cp_le_reject_cis *cp;
+	struct net_buf *buf;
+	int err;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_REJECT_CIS, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(handle);
+	cp->reason = reason;
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_REJECT_CIS, buf, NULL);
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+int hci_le_accept_cis(uint16_t handle)
+{
+	struct bt_hci_cp_le_accept_cis *cp;
+	struct net_buf *buf;
+	int err;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_ACCEPT_CIS, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(handle);
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_ACCEPT_CIS, buf, NULL);
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+void hci_le_cis_req(struct net_buf *buf)
+{
+	struct bt_hci_evt_le_cis_req *evt = (void *)buf->data;
+	uint16_t acl_handle = sys_le16_to_cpu(evt->acl_handle);
+	uint16_t cis_handle = sys_le16_to_cpu(evt->cis_handle);
+	struct bt_conn *conn, *iso;
+	int err;
+
+	BT_DBG("acl_handle %u cis_handle %u cig_id %u cis %u",
+		acl_handle, cis_handle, evt->cig_id, evt->cis_id);
+
+	/* Lookup existing connection with same handle */
+	iso = bt_conn_lookup_handle(cis_handle);
+	if (iso) {
+		BT_ERR("Invalid ISO handle %d", cis_handle);
+		hci_le_reject_cis(cis_handle, BT_HCI_ERR_CONN_LIMIT_EXCEEDED);
+		bt_conn_unref(iso);
+		return;
+	}
+
+	/* Lookup ACL connection to attach */
+	conn = bt_conn_lookup_handle(acl_handle);
+	if (!conn) {
+		BT_ERR("Invalid ACL handle %d", acl_handle);
+		hci_le_reject_cis(cis_handle, BT_HCI_ERR_UNKNOWN_CONN_ID);
+		return;
+	}
+
+	/* Add ISO connection */
+	iso = bt_conn_add_iso(conn);
+
+	bt_conn_unref(conn);
+
+	if (!iso) {
+		hci_le_reject_cis(cis_handle,
+				  BT_HCI_ERR_INSUFFICIENT_RESOURCES);
+		return;
+	}
+
+	/* Request application to accept */
+	err = bt_iso_accept(iso);
+	if (err) {
+		bt_iso_cleanup(iso);
+		hci_le_reject_cis(cis_handle,
+				  BT_HCI_ERR_INSUFFICIENT_RESOURCES);
+		return;
+	}
+
+	iso->handle = cis_handle;
+	iso->role = BT_HCI_ROLE_SLAVE;
+	bt_conn_set_state(iso, BT_CONN_CONNECT);
+
+	hci_le_accept_cis(cis_handle);
+}
+
+int hci_le_remove_cig(uint8_t cig_id)
+{
+	struct bt_hci_cp_le_remove_cig *req;
+	struct net_buf *buf;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_REMOVE_CIG, sizeof(*req));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	req = net_buf_add(buf, sizeof(*req));
+
+	memset(req, 0, sizeof(*req));
+
+	req->cig_id = cig_id;
+
+	return bt_hci_cmd_send_sync(BT_HCI_OP_LE_REMOVE_CIG, buf, NULL);
+}
+
+void bt_iso_cleanup(struct bt_conn *conn)
```

Comment:
This function seems to be valid for CIS only. Perhaps rename to `bt_iso_cleanup_cis` or use handle as argument instead of a `bt_conn`?

Reply:
Will change it to just take the handle, I guess any command which applies to BIS as well will need to map to handle, we could in theory make a BIS map to a bt_conn as well, since the ISO packets are also meant to carry broadcast data.

Reply:
> we could in theory make a BIS map to a bt_conn as well
I was wondering the same thing. Would be great if we somehow could use the same functions for sending data over BIS as we do for CIS.